### ISSUE TITLE
Is this Maintained?

### ISSUE THREAD

**Issue by Trickfilm400** [UnauthorizedAccessBV/ESET-Protect-Docker-Server#12]
I am interessted in an updated version for version 11 of the eset on prem server, because the centos support is ending.

But this seems not that up to date - are maintainers needed?

I would be able to update some parts here and there for the update to version 11, but only if this would be implemented

**Comment by dmaasland** [comment]
Heh, I wasn't aware that there were people actually using this. I'll put together an updated version :)

**Comment by Trickfilm400** [comment]
Yeah, I searched for a solution, because the current on-prem eset protect server will be end-of-life (https://support-eol.eset.com/en/business_managed_remote_va.html) due to the centos 7 support end at june 30, 2024

so I am looking forward for a docker compose setup with traefik

Thanks for the updates, maybe I can help later on

**Comment by dmaasland** [comment]
I've updated botth the Server and Console images to v11. Please let me know if it works for you!

**Comment by Trickfilm400** [comment]
Thanks, I will try it out, but it can take a bit, I need to prepare this before for the migration of the database and so on

But I will definitely let you know

**Comment by Trickfilm400** [comment]
Got it working, the migration was a bit difficult, but the docker works perfectly

----
Another question is the update procedure, the updater of the server itself will not work due to the docker, so the docker has to be updated?

Is there a way to get update notifications or is this in the server itself displayed, do you know this by any chance?

**Comment by abysso2** [comment]
@Trickfilm400 : Did you manage to connect the agents - and if, how? :-D

https://github.com/UnauthorizedAccessBV/ESET-Protect-Docker-Server/issues/13

**Comment by dmaasland** [comment]
Sorry, it's been crazy busy the past few weeks. There is a repository which you can check, but it's a binary format and I haven't looked at it yet:

- https://repository.eset.com/v1/info.meta
- https://repository.eset.com/v1/com/eset/apps/business/era/server/linux/metadata
- https://repository.eset.com/v1/com/eset/apps/business/era/webconsole/metadata

Anyway, I've updated the images to v12 so they should appear in a few minutes

**Comment by Trickfilm400** [comment]
Oh wow, what a timing :D - Thanks for the update!
I spent way to many hours updating it myself yesterday, as I planned a PR.

> I have updated the image in the last locally to 11.1, but I wanted to update the Ubuntu base to 24.04 and I'm having issues with the PDF Report thing. After figuring out all dependencies it complained about no display (`no $DISPLAY or X Server running ...`) and that there is no dbus running, bla bla.
> With the old included shell file it will spawn the process with a few sub-processes and just hang from there and nothing is happening. I mean I don't need the report generation, but I don't know why it's broken with Ubuntu 24.04 (maybe also the current version? I didn't checked before)

The current state of my PDF thing is a error message: `Failed to generate report: Object not present in object storage` and in the `trace.log` file `CMetadataProcessorV3: Downloading file from http://repository.eset.com/v1/com/eset/eulas/product/lg/eei/metadata3 has failed 3 times with: GetFile: Object 'http://repository.eset.com/v1/com/eset/eulas/product/lg/eei/metadata3' not found [error code: 20011]`
Maybe its a bug with eset itself, as yesterday the repository.eset.com/v1 gave me 404 errors and I used the download.eset.com URL..

TLDR: If the report generation somehow works again, I could create a PR for ubuntu 24 or tell you the dependencies if you are interested.

-----

To the repository files: I opened them with VSC and there is a signing certificate at the top and bottom and between there is just a JSON object: Here's a part of the json snippet:
```json
    {
      "Version": "12.0.273.0",
      "VersionID": 3377699738419200,
      "Date": 1733823819,
      "Hash": "1fb1d8216c720c0db3b45761ed91fcb8a39c13c8",
      "Size": 493280033,
      "DisplayName": "ESET PROTECT On-Prem Server",
      "OSType": "linux",
      "OSNames": "Debian,Ubuntu, Red Hat, Hat Enterprise Linux",
      "Language": "en_US",
      "Description": "ESET PROTECT On-Prem Server",
      "Legacy": false,
      "SoftwareFamily": "ESET Remote Administrator",
      "Path": "v12/12.0.273.0/server_linux_x86_64.sh",
      "Changelog": "v12/12.0.273.0/server_linux_x86_64.sh.changelog.html",
      "Eula": "v12/12.0.273.0/Server-Linux-x86_64.sh.eula/eula_en-us.html"
    }
```

So we need a method to remove/verify the certificate of the file and parse then the json in the file, as the versions are in an array, it should be pretty easy to check for new versions

**Comment by dmaasland** [comment]
Sure, let me know what 24.04 needs, pr is fine to :)

---
I forgot I actually had code for this lying around: https://gist.github.com/dmaasland/57f03cd7ed5af910f7b141832c16222b . The signing stuff is just GPG so pretty easy to parse. Unfortunately there isn't really a clear "latest" tag anywhere. Here is the output:

## Server
https://gist.github.com/dmaasland/3d99474d42a8025cce651f28b06293cf

## Console
https://gist.github.com/dmaasland/c6662d3edcfed4efc785d18833c38986

**Comment by Trickfilm400** [comment]
For 24.04 I found these dependencies:
All after `lshw` are new and `libqtwebkit4` isn't needed anymore
```Dockerfile
RUN apt update && apt install -y --no-install-recommends \
    xvfb \
    ca-certificates \
    cifs-utils \
    krb5-user \
    ldap-utils \
    libsasl2-modules-gssapi-mit \
    snmp \
    libodbc2 \
    libodbcinst2 \
    libssl-dev \
    samba \
    python3 \
    lshw \
    libglib2.0-0 \
    libnss3 \
    libatk1.0-0t64 \
    libatk-bridge2.0-0t64 \
    libxcomposite1 \
    libxdamage1 \
    libgbm1 \
    libcairo2 \
    libxkbcommon0 \
    libpango-1.0-0 \
    libasound2t64 \
    && rm -rf /var/lib/apt/lists/*
```

This is no longer needed AFAIK:
```Dockerfile
RUN apt-get update && apt-get install -y --no-install-recommends \
    software-properties-common \
    gpg-agent \
    && add-apt-repository ppa:ubuntuhandbook1/ppa
```

Otherwise it seems the same - as long as it would work - maybe on your side its working

-----

Oh, GPG keys are a thing, I totally forgot and was confused, as it was too short for an RSA key... Well

What you could do maybe is get the `version_id` and then check for all other versions the `update` key and check, if the version is higher than the other ones. If the `update` key includes the same version id as we are checking, it is the latest option. So like building a virtual upgrade path and figuring out the latest version by this?

Edit: You would need to merge all `update` objects into one with the correct order (as not every version contains the full upgrade path for all included versions in this object and therefore fails to work correctly when searching) and then get the top one as the latest. But besides this, is the latest version nut the one with the highest `version_id`? It doesn't seem there are multiple versions supported at once, so I think the easiest option is to use the highest `version_id` as the latest version, as this was always the case in the past?
